expect/_build_message.ts

```diff
@@ -43,19 +40,8 @@ export function buildNotEqualErrorMessage<T>(
   options: EqualErrorMessageOptions,
 ): string {
   const { msg } = options || {};
-  let actualString: string;
-  let expectedString: string;
-
-  try {
-    actualString = String(actual);
-  } catch {
-    actualString = CAN_NOT_DISPLAY;
-  }
-  try {
-    expectedString = String(expected);
-  } catch {
-    expectedString = CAN_NOT_DISPLAY;
-  }
+  const actualString = String(actual);
+  const expectedString = String(expected);
```

Comment:
[`String()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/String) as a function is extremely versatile. I can't think of any objects that would throw when passed as input to it unless it's something super contrived like:
```ts
const problematicObject = {
  toString() {
    throw new Error("Cannot convert to string");
  }
};
```

Reply:
> ```ts
> const problematicObject = {
>   toString() {
>     throw new Error("Cannot convert to string");
>   }
> };
> ```

So the `catch` block is not unreachable with the above example. I don't think we should remove it

Reply:
`String()` would only fail if an object deliberately throws on `.toString()`. I don't know of any other cases in which it'd throw. If `String()` does throw, does changing the message to `[Cannot display]` help with troubleshooting?

Reply:
> `String()` would only fail if an object deliberately throws on `.toString()`.

I think that can be also caused by mistake. For example, some might define `toString` like:

```ts
  toString() {
    return JSON.stringify(this.someProperty);
  }
```

This throws when someProperty includes BigInt or circular reference ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#exceptions))

Reply:
> If `String()` does throw, does changing the message to `[Cannot display]` help with troubleshooting?

`[Cannot display]` is actually not helpful. I think we should instead append the thrown error message after it like `Cannot display because of the error: ${originalErrorMessage}`

Reply:
Does changing the message to `[Cannot display]` help in troubleshooting in the case you provided, or is it better to throw?

Reply:
Why not just let it throw?

Reply:
Sounds a fair point. Probably it's a little better to let it throw as is than the current state (The current `cannot display` message only hides what the actual error is).